pkg/query-service/app/parser.go

```diff
@@ -886,6 +886,14 @@ func parseFilterAttributeValueRequest(r *http.Request) (*v3.FilterAttributeValue
 		return nil, err
 	}
 
+	if err := tagType.Validate(); err != nil {
+		return nil, err
+	}
+
+	if err := filterAttributeKeyDataType.Validate(); err != nil {
+		return nil, err
+	}
```

Comment:
They were not validated here because there could be cases where these two are unavailable at the client's request.

Reply:
`filterAttributeKeyDataType` can be missing but `tagType` will be available at client's request

Reply:
Both may not be present, it is not necessary that the user must pick the key from the API suggestions only because they are limited to 2 days, and the key they are interested in doesn't exist in the suggestions; in that case, they will write their key-operator-value in the criteria, and client request will not have the `tagType` or its data type because the client doesn't have any metadata about them. This was highlighted several times, including in logs PR.

Reply:
Ok, I will return empty response if any of them are not present.